File: 01/1.php
<?php

declare(strict_types=1);

$strings = file('test.txt');

$numbers = array_map(function (string $string): int {
    $number = preg_replace("/[^0-9]/", "", $string);
    return (int) ($number[0].$number[strlen($number)-1]);

}, $strings);

echo array_sum($numbers).PHP_EOL;

File: 01/1.py
import re

f = open("test.txt", "r+")
rows = f.read().splitlines()

sum = 0
for row in rows:
    number = re.sub('[^0-9]', '', row)
    number = number[0]+number[len(number)-1]
    sum += int(number)
    
print(sum)

    
